lib/charms/prometheus_k8s/v0/prometheus_scrape.py

```diff
@@ -335,6 +339,81 @@ def _on_scrape_targets_changed(self, event):
 }
 
 
+RELATION_INTERFACE_NAME = "prometheus_scrape"
+
+
+class RelationDirection(Enum):
+    """Relations can be either provided or required by a charm."""
+
+    PROVIDED = 1
+    REQUIRED = 2
+
+
+class NoRelationWithInterfaceFoundError(Exception):
+    """No relations with the given interface are found in the charm meta."""
+
+    def __init__(self, charm: CharmBase, relation_interface: str):
+        self.charm = charm
+        self.relation_interface = relation_interface
+        self.message = (
+            f"No relations with interface '{relation_interface}' found in the meta "
+            f"of the '{charm.meta.name}' charm"
+        )
+
+        super().__init__(self.message)
+
+
+class MultipleRelationsWithInterfaceFoundError(Exception):
+    """Multiple relations with the given interface are found in the charm meta."""
+
+    def __init__(self, charm: CharmBase, relation_interface: str, relations: list):
+        self.charm = charm
+        self.relation_interface = relation_interface
+        self.relations = relations
+        self.message = (
+            f"Multiple relations with interface '{relation_interface}' found in the meta "
+            f"of the '{charm.name}' charm"
+        )
+
+        super().__init__(self.message)
+
+
+def _get_single_relation_by_interface(
```

Comment:
This method is only used by Provider. So what is the motivation to put this in Module scope ?

Reply:
No also Consumer now

Reply:
Consider renaming to `_get_relation_by_interface`: the `single` is redundant.